ateam_kenobi/src/in_play_eval.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "in_play_eval.hpp"
+
+#include <CGAL/squared_distance_2.h>
+#include <cmath>
+#include <limits>
+
+namespace ateam_kenobi
+{
+
+void InPlayEval::Update(World & world)
+{
+  const auto game_command = world.referee_info.running_command;
+  if (game_command == ateam_common::GameCommand::ForceStart) {
+    in_play_ = true;
+  } else if (IsGameStopped(world)) {
+    in_play_ = false;
+    ball_start_pos_.reset();
+  } else if (IsStopCommandEnding(world)) {
+    SetTimeout(world);
+    SetDistanceThreshold(world);
+  } else if (IsGameResuming(world)) {
+    ball_start_pos_ = world.ball.pos;
+    timeout_start_ = std::chrono::steady_clock::now();
+  } else if (HasTimeoutExpired()) {
+    in_play_ = true;
+  } else if (HasBallMoved(world)) {
+    in_play_ = true;
+  }
```

Comment:
```suggestion
  } else {
      in_play_ = false;  
  }
```
In case we hit an edge case that violates all of these conditions, do you think it makes sense to set `in_play_` to false?

Reply:
No, the idea with this function is that all of the cases checked are when `in_play_` needs to change. If none of the conditions are hit, we just need to keep the same value we had from the last frame.